Fifth homework/JS_HW_5.js

```diff
@@ -0,0 +1,101 @@
+  - Например: mergeArrays([1,2], [3,4], [5,6]) // [1,2,3,4,5,6]
+  - Решить с использованием Spread operator */
+
+  function mergeArrays(...arrFunc) { // раз спред - всегда спред)
+    const numArray = []; // в новый массив будем скаладывать наши входящие массивы
+    for (const arr1 of arrFunc) numArray.push(...arr1); //перебираем входящий массив с массивами и пушим спредом в наш созданный. решил попробывать соркащенное написание))
+    return numArray;
+  }
+  pr(mergeArrays([1, 2], [3, 4], [5, 6]));
+  pr(mergeArrays([1, 2], [3, 4], [5, 6],[7, 8], [9, 10], [11, 12]));
+  line();
+
+/* 2. Devide by _
+  - Написать функцию, которая преобразует любое предложение в вот_Такой_Вот_Вид и возвращает его. 
+  - Первое слово должно начинаться с буквы в нижнем регистре, у остальных -  верхнем. 
+  - Пример: I am super engineer => i_Am_Super_Engineer */
+
+  function devideBy(a = "") { // сразу переменную в текст
+    let textEdit = a.replaceAll(" ", "_").toLowerCase(); // меняем старое на новое и всё в низ - условие с первым словом в кармане)
+    for (i = 0; i < textEdit.length; i++)
+       if (textEdit[i - 1] === "_") textEdit[i].toUpperCase(); // опять сокращаю) по "_" хитрим и меняем символы вверх.
+    return textEdit;
+  }
+  pr(devideBy("ПривеТ! Я коТлеТка с МЯСом)")); // текст диктовал сын))
+  pr(devideBy("I am super engineer"));
+  line();
+
+// /* 3. Фибаначчи
+//   - Напишите функцию fibanacci(n), возвращающую энное число Фибоначчи
+//   - числа Фибоначчи (строка Фибоначчи) — числовая последовательность, первые два числа которой являются 0 и 1, а каждое последующее за ними число является 
+//   суммой двух предыдущих
+//   - Например fibanacci(8) //21 */
+
+function fibanacci(n) {
+  const fibArr = []; // массив куда складываем ряд Фибоначчи. 
+  let nFibRet = 0; 
+  for (i = 0; i <= n; i++)
+    i == 0 || i == 1 ? fibArr[i] = i : fibArr[i] = fibArr[i - 2] + fibArr[i - 1]; // Перебираем ряд Фибоначи и заносим в массив.
+    nFibRet = +(fibArr.slice(-1)); // Вырезаем последнее значение в массиве и кладем в переменную). 
+  return nFibRet;
+}
+pr(fibanacci(8));
+pr(fibanacci(10));
+line();
+
+/* 1. Написать функцию, которая принимает на вход слово и проверяет, является ли это слово палиндромом */
+
+function palindromCheck(p = "") {
+  const palinLow = p.toLowerCase(); // а мало ли) чтобы БуКаВки все были одного роста)
+  const palinReverse = palinLow.split('').reverse().join(''); // разделяем, меняем порядок символов на обратный и склеиваем в слово обратно
+  palinReverse.localeCompare(palinLow) == 0 ? pr("палиндром") : pr("хер тоби") /* ну тут все и так ясно) полное совпадение прямого и отражения - значит палиндром.
+                                                                                решил сразу выводить через консоль.*/ 
+  return palinReverse; // не.. можно конечно и в еще одну перменнуб выводить значение, что в консоли. пока оставил так)
```

Comment:
Возвращаем true || false, достаточно
```suggestion
  return  palinLow.split('').reverse().join('') === palinLow;
```